src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -0,0 +1,3532 @@
+            newMinimum.temp      = currentT;
+            newMinimum.potential = V(zeroTemp) * (1 + currentT * currentT);
+
+            if (old_min_is_global) // check if newMinimum is
+                                   // still global minimum
+            {
+              IsGlobMin(newMinimum);
+              if (!newMinimum.is_glob_min)
+              {
+                if (output)
+                  Logger::Write(
+                      LoggingLevel::MinTracerDetailed,
+                      "Phase no longer coincides with global minimum at " +
+                          std::to_string(newMinimum.temp));
+                globMinEndT       = newMinimum.temp;
+                old_min_is_global = false;
+              }
+            }
+
+            MinimumList.push_back(newMinimum);
+          }
+          break;
+        }
+        else
+        {
+          newMinimum.point     = new_point;
+          newMinimum.temp      = currentT;
+          newMinimum.potential = V(new_point) * (1 + currentT * currentT);
+
+          if (old_min_is_global) // check if newMinimum is
+                                 // still global minimum
+          {
+            IsGlobMin(newMinimum);
+            if (!newMinimum.is_glob_min)
+            {
+              if (output)
+                Logger::Write(
+                    LoggingLevel::MinTracerDetailed,
+                    "Phase no longer coincides with global minimum at " +
+                        std::to_string(newMinimum.temp));
+              globMinEndT       = newMinimum.temp;
+              old_min_is_global = false;
+            }
+          }
+
+          MinimumList.push_back(newMinimum);
+          // Sucess minimum!
+          ss << "\033[1;32m.\033[0m";
+          dT *= .5 * ThresholdDistance /
+                Distance; // Try to predict the best stepsize
+          if (abs(initialdT) <= abs(dT)) dT = initialdT;
+        }
+        IsInMin = -1;
+      }
+      point = new_point;
+    }
+
+    // Make sure that or step is not bigger than it should be and we overshot
+    // the next minimum
+    if (currentT == finalT) break;
+    bool SafeStep = abs(finalT - currentT) > abs(dT);
+    if (SafeStep)
+      currentT += dT;
+    else
+      currentT = finalT;
+  }
+  if (output) Logger::Write(LoggingLevel::MinTracerDetailed, ss.str());
+  if (output)
+    Logger::Write(LoggingLevel::MinTracerDetailed,
+                  "T = " + std::to_string(currentT) + " GeV");
+  if (old_min_is_global) globMinEndT = currentT;
+  return MinimumList;
+}
+
+/**
+ * @brief TrackPhase
+ * @param point start point for tracking
+ * @param currentT start point temperature for phase tracking
+ * @param finalT end point temperature
+ * @param dT initial temperature step size
+ * @param output if true tracking output is printed on the screen
+ * @param unprotected if true we dont check the hessian
+ */
+std::vector<Minimum> MinimumTracer::TrackPhase(std::vector<double> point,
```

Comment:
```suggestion
std::vector<Minimum> MinimumTracer::TrackPhase(const std::vector<double>& point,
```

Reply:
This cannot be implemented. We use/change ```point``` in the code.

Reply:
Done.